### DISCUSSION TITLE
NanoEls Build

### DISCUSSION THREAD

**Discussion by zabu83** [kachurovskiy/nanoels#11] [General]
First of all I want to thank Maxim Kachurovskiy for his great work.

For my construction I have simply followed all the steps of maxim and solving the problems that I found.

First I started with the flange for the nema 23 stepper motor. 
![MVIMG_20210417_210901](https://user-images.githubusercontent.com/82053978/121730223-ae961d00-caef-11eb-889a-0d323ea8bcf2.jpg)

I made the flange with 12mm steel on the lathe and trimmed with a bandsaw. 
I used the plywood jig but it didn't seem to fit quite right and I had to cut it even more. 
![MVIMG_20210417_192521](https://user-images.githubusercontent.com/82053978/121731269-008b7280-caf1-11eb-9ee3-cae5be28329d.jpg)

The motor I used a closed loop nema 23 (stepperonline). It is a little bigger and I had to modify the case because it did not fit. 
![IMG_20210611_201927](https://user-images.githubusercontent.com/82053978/121733228-83adc800-caf3-11eb-8027-27eba052dcaf.jpg)

the motor shaft is 8mm instead of 10mm and it is a little shorter than the motor of the maxim motor, so the pulley bolt did not catch the shaft and I had to open a new threaded hole more on the edge.

![IMG_20210611_201832](https://user-images.githubusercontent.com/82053978/121733298-97f1c500-caf3-11eb-845b-92f89e3cbfff.jpg)
![IMG_20210611_201855](https://user-images.githubusercontent.com/82053978/121733318-9b854c00-caf3-11eb-943b-6b14ce9eac51.jpg)

Finally, in order not to make the post so long, I am going to leave some various images.
If you have any questions about my build, don't hesitate to ask.

![IMG_20210611_201906](https://user-images.githubusercontent.com/82053978/121734198-be643000-caf4-11eb-889e-3d78794db003.jpg)
![MVIMG_20210420_174104](https://user-images.githubusercontent.com/82053978/121734212-c1f7b700-caf4-11eb-9e5f-76637c238876.jpg)
![MVIMG_20210425_202929](https://user-images.githubusercontent.com/82053978/121734230-c7550180-caf4-11eb-8a0a-3ca8bd53404d.jpg)
![MVIMG_20210501_204000](https://user-images.githubusercontent.com/82053978/121734253-ccb24c00-caf4-11eb-9082-b8e319c17ea3.jpg)
![MVIMG_20210502_154703](https://user-images.githubusercontent.com/82053978/121734265-d20f9680-caf4-11eb-9029-116fda818436.jpg)
![MVIMG_20210503_192931](https://user-images.githubusercontent.com/82053978/121734274-d63bb400-caf4-11eb-855a-f9391ae470ff.jpg)

Again a big thank you to Maxim for the great job.

PS: To say that the threading is very good and does not lose steps, I am very very happy.

**Comment by kachurovskiy** [comment]
Congrats and thank you for the kind words!

**Reply by zabu83** [reply]
thank you. I have ordered a nema 17 motor to control the cross slider, I have printed the bracket for the motor and bought the bearings. I had thought about controlling it with a slider and a mini controller, but I was wondering if it is possible to use your code at least only to control the advance of the cross slider to be able to face.

**Reply by kachurovskiy** [reply]
My friend, if you'd like to build a 2-axis ELS, please see the files attached below for the code and instructions. It's based on Arduino Mega though. Source - https://www.chipmaker.ru/topic/118083/

[Digital_Feed_7e2.zip](https://github.com/kachurovskiy/nanoels/files/6643873/Digital_Feed_7e2.zip)

**Reply by zabu83** [reply]
I have been seeing it but the truth is that I am not able to understand it. 

I would only like to have the advance function on the second axis. 

Its build is much simpler and it would be great to be able to control the second axis with its build.

**Reply by kachurovskiy** [reply]
zabu83, sadly I won't be able to provide a working 2-axis version of NanoEls any time soon.

With Digital_Feed_7e2 you can do so much more - cut threads in multiple steps, automatic multi-step facing in both directions, even spheres. You can also keep it simple and only use the few options you need.

> I am not able to understand it.

Please let me know which part exactly you can't understand. For source code comments in Russian, use translate.google.com.

I'm looking to create a new GitHub repository with Digital_Feed_7e2 translation and instructions but I'm waiting to hear from the author first.

**Reply by zabu83** [reply]
Well, sincerely I don't understand practically anything, it is very confusing for me even using the translator.

I don't know exactly what parts are needed for the construction, I don't know if the ones I already have, such as the encoder, can be of use to me, and I don't know how to understand the diagrams either.

It would be genail to see a translation of Digital_Feed_7e2.

I still think that its build is great and also the explanation of the construction is very clear.

**Comment by zabu83** [comment]
hi maxim, can you help me with the scheme? Can you tell me how the connections are going with any drawing?

I leave you my email in case you don't want to mix topics here.

Thank you so much.

**Reply by kachurovskiy** [reply]
Hey zabu83, you should be able to read Oleg's scheme once you get a hang of the basic principles - please take a look at https://learn.sparkfun.com/tutorials/how-to-read-a-schematic/all which provides a good and concise overview.

![!_ELS_V7](https://user-images.githubusercontent.com/517919/122011310-3dd75500-cdbc-11eb-8c81-30da0fdb8295.jpg)
Source: https://www.chipmaker.ru/topic/118083/

The most confusing part of this scheme is the top-left but you can ignore that since you already have a rotary encoder (that's what it's showing). All this scheme is showing is how you connect external components (buttons, motors, buzzer, display, ...) to the Mega board. It also specifies ~30 pull-up resistors because Arduino built-in resistors are too weak.

I'm working on a PCB that you will be able to order on jlcpcb.com for ~15$ that will go between Arduino Mega and a display. It will also serve as a mounting board for all the buttons and connections and make implementing this scheme very easy:

<img width="955" alt="df1" src="https://user-images.githubusercontent.com/517919/122011621-94dd2a00-cdbc-11eb-9606-7336e54447ef.png">

The list of components to be mounted in this board is:

- A lot of buttons
- 4 resistor arrays containing 8 1kOhm resistors each to save space instead of using standalone resistors
- LEDs for limit indication
- Places for Mega and display to be plugged into
- Buzzer
- Potentiaometers to adjust display brightness and feed speed
- Screw terminals to connect things like motors

Here's how it will look in 3D:

<img width="1248" alt="df2" src="https://user-images.githubusercontent.com/517919/122011995-fe5d3880-cdbc-11eb-94d3-cbb2f5a995ef.png">

It will take some time to finish this PCB, order, assemble and test it. I'll be away in August and not sure if I'll be able to finish before that. Feel free to wait or proceed with your own implementation.

Best wishes!

**Reply by zabu83** [reply]
Thanks for the info.

That pcb is great, it will be worth the wait.

keep me informed, anything I can help you do not hesitate to tell me.

on the other hand the nema 17 motor is enough for the second axis?
Would it also be necessary for the motor to be closed loop?

**Reply by kachurovskiy** [reply]
My PCB turned out to have a bunch of mistakes and I also should have used 4 rotary switches instead of 4 of those button groups. It's definitely going into September / October since I need to get parts from China and then design the v2 PCB and I'm out in August.

For the cross-slide, I have NEMA 17 geared down (1 rotation of the motor - 1/3 rotation of the axis) and it's very much enough. Admittedly, I didn't test it for things like cutting off where a lot of force is required. Closed loop seems optional though of course it's better. It will definitely be much harder to mount a closed loop motor there than open loop.

![PXL_20210710_192819708](https://user-images.githubusercontent.com/517919/125174446-dbf4ea00-e1c5-11eb-8984-3a80bd88c9a4.jpg)

Playing with the 2-axis Russian Digital Feed project a little bit and it's mostly working. It's rather hard to use though and I'm still having a lot of unexplained problems (cutting air for now to avoid equipment damage).

Ordered a 270 EUR 2-axis ELS from https://www.rocketronics.de yesterday. I think it will be a better option for most people, has more functions and not that much more expensive than building the Russian Digital Feed project. Depending on the components you'd use for the DIY build, it can cost 100 EUR just in parts and then weeks for assembly and debugging.

How's it going in your shop? Is the ELS still working? :) Cheers!

**Reply by zabu83** [reply]
Glad to see your progress even though there are bugs.
the buid that you have mounted in the photo looks very good and is quite reduced.

the nema 17 motor, seeing in its video that it is not capable of stopping it with its force, I think it is enough.
I have to order a new nema 17 engine because it got lost on the road.
Is there a better option to the dm542t stepperonline controller for mena 17?

rocketronic els is a very good option to avoid complicating your life, although like me, I think most people like the adventure of setting up an els project like yours.
for me it is 100% the part that I enjoy the most about this hobby.

Edit: ELS still works perfectly. I am very happy, but it only has one axis ;)


If you continue with that project, I look forward to seeing you.
keep me informed

greetings friend.

**Comment by Diez-10** [comment]
Hola zabu83
Estoy construyendo NanoEls
Tengo interes en Digital_feed7e2.zip y su PCB

**Reply by zabu83** [reply]
Hello, the author of this project is @kachurovskiy

**Comment by Diez-10** [comment]
Hola Zabu83
Gracias por la aclaracion
